From b8e6c051c6ad6ae3ae93db8e74f37dbc82c22514 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 17 Aug 2016 16:25:09 +0200
Subject: [PATCH] network:ibft put IPv6 IP in brackets

ip=2620:0052:0000:2220:0226:b9ff:fe81:cde4::[2620:0052:0000:2220:0000:0000:0000:03fe]:64::ibft0:none

should be

ip=[2620:0052:0000:2220:0226:b9ff:fe81:cde4]::[2620:0052:0000:2220:0000:0000:0000:03fe]:64::ibft0:none

https://bugzilla.redhat.com/show_bug.cgi?id=1322592#c19
---
 modules.d/40network/net-lib.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index de334d98..3459e898 100755
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -277,6 +277,7 @@ ibft_to_cmdline() {
                 [ -e ${iface}/hostname ] && hostname=$(read a < ${iface}/hostname; echo $a)
                 if [ "$family" = "ipv6" ] ; then
                     if [ -n "$ip" ] ; then
+                        ip="[$ip]"
                         [ -n "$prefix" ] || prefix=64
                         mask="$prefix"
                     fi
